tests/CommandIsolatorTest.cpp

```diff
@@ -90,9 +90,18 @@ TEST_F(CommandIsolatorSimpleTest,
 }
 
 TEST_F(CommandIsolatorSimpleTest,
-       should_run_isolate_command_and_terminate_successfully) {
+       should_run_isolate_command_without_context_and_terminate_successfully) {
+  isolator->cleanContainerContext(containerId);
```

Comment:
I think this test can be removed if it makes you change the scope of `cleanContainerContext` jsut to make this test (I see you changed it from private to public).

The second newly added test is sufficient imo